sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -137,10 +140,12 @@ private[sql] object FileSourceStrategy extends Strategy with Logging {
 
           val splitFiles = selectedPartitions.flatMap { partition =>
             partition.files.flatMap { file =>
+              val blockLocations = getBlockLocations(file)
               (0L to file.getLen by maxSplitBytes).map { offset =>
                 val remaining = file.getLen - offset
                 val size = if (remaining > maxSplitBytes) maxSplitBytes else remaining
-                PartitionedFile(partition.values, file.getPath.toUri.toString, offset, size)
+                val hosts = getBlockHosts(blockLocations, offset)
+                PartitionedFile(partition.values, file.getPath.toUri.toString, offset, size, hosts)
```

Comment:
How about we add document to this FileSourceStrategy to explain our strategy? So, readers can quickly understand what we are doing at here.